src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -567,6 +569,23 @@ private async Task PackageSourcesChangedAsync(IReadOnlyCollection<PackageSourceC
             }
         }
 
+        private async Task IsCentralPackageManagementEnabledAsync(CancellationToken cancellationToken)
+        {
+            await NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async delegate
+            {
+                // Go off the UI thread to perform non-UI operations
+                await TaskScheduler.Default;
+
+                if (!Model.IsSolution)
```

Comment:
An optimization would be to perform this check before invoking this method at all, or at the top of the method. Don't want Solution PMUI switching threads unnecessarily.

Reply:
I prefer this approach. What do you think?
https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs#L210